File: 0057-insert-interval/0057-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        intervals.push_back(newInterval);
        sort(intervals.begin(), intervals.end());

        vector<vector<int>>ans;
        int i=0;
        int n=intervals.size();

        

        while(i<n)
        {
            int startI=intervals[i][0];
            int endI=intervals[i][1];
            for(int j=i+1;j<n;j++)
            {
                if(intervals[j][0]>=startI && intervals[j][0]<=endI)
                {
                    endI=max(endI,intervals[j][1]);
                    i=j;
                }
            }
           

            ans.push_back({startI,endI});
            i++;
            
        }
        return ans;
    }
};

File: 0057-insert-interval/0057-insert-interval.py
class Solution(object):
    def insert(self, intervals, newInterval):
        """
        :type intervals: List[List[int]]
        :type newInterval: List[int]
        :rtype: List[List[int]]
        """

        ans=[]

        if(len(intervals)==0):
            ans.append(newInterval)
            return ans
        
        intervals.append(newInterval)


        intervals.sort()
        
        s=intervals[0][0]
        e=intervals[0][1]

        n=len(intervals)
        for i in range(1,n):
            if(s<=intervals[i][0] and e>=intervals[i][0]):
                s=min(s,intervals[i][0])
                e=max(e,intervals[i][1])
                
            else:
                
                lis=[]
                lis.append(s)
                lis.append(e)
                ans.append(lis)

                s=intervals[i][0]
                e=intervals[i][1]

        lis=[]
        lis.append(s)
        lis.append(e)
        ans.append(lis)

        return ans

            

                

                
            
            
                
